packages/react/src/checkbox/indicator/CheckboxIndicator.tsx

```diff
@@ -22,19 +27,57 @@ const CheckboxIndicator = React.forwardRef(function CheckboxIndicator(
   props: CheckboxIndicator.Props,
   forwardedRef: React.ForwardedRef<HTMLSpanElement>,
 ) {
-  const { render, className, keepMounted = true, ...otherProps } = props;
+  const { render, className, keepMounted = false, animated = true, ...otherProps } = props;
 
-  const state = useCheckboxRootContext();
+  const rootState = useCheckboxRootContext();
 
-  const customStyleHookMapping = useCustomStyleHookMapping(state);
+  const rendered = rootState.checked || rootState.indeterminate;
+
+  const { mounted, transitionStatus, setMounted } = useTransitionStatus(rendered);
+
+  const indicatorRef = React.useRef<HTMLSpanElement | null>(null);
+  const mergedRef = useForkRef(forwardedRef, indicatorRef);
+
+  const hidden = !mounted;
+
+  const state: CheckboxIndicator.State = React.useMemo(
+    () => ({
+      ...rootState,
+      transitionStatus,
+      hidden,
+    }),
+    [rootState, hidden, transitionStatus],
+  );
+
+  useAfterExitAnimation({
+    open: rendered,
+    animated,
+    animatedElementRef: indicatorRef,
+    onFinished() {
+      setMounted(false);
+    },
+  });
+
+  const baseStyleHookMapping = useCustomStyleHookMapping(rootState);
+
+  const customStyleHookMapping: CustomStyleHookMapping<CheckboxIndicator.State> = React.useMemo(
+    () => ({
+      ...baseStyleHookMapping,
+      ...transitionStatusMapping,
+    }),
+    [baseStyleHookMapping],
+  );
 
   const { renderElement } = useComponentRenderer({
     render: render ?? 'span',
-    ref: forwardedRef,
+    ref: mergedRef,
     state,
     className,
     customStyleHookMapping,
-    extraProps: otherProps,
+    extraProps: {
+      hidden,
+      ...otherProps,
+    },
```

Comment:
One consequence of `hidden` here is that you can't rely on the outer `Root` checkbox animation, because it looks for animations on the indicator itself: `animatedElementRef: indicatorRef`. So you need to duplicate styles to keep it mounted; `hidden={false}` of course can be used